src/core/ledger/currency.js

```diff
@@ -53,32 +53,48 @@ const protocolSymbolParser: C.Parser<ProtocolSymbol> = C.exactly([
  * Chains like Bitcoin and Filecoin do not have "production" sidechains so
  * we represent them as a string, as specified in the ProtocolSymbol type
  */
-export type ProtocolId = {|
+export type Protocol = {|
   +type: "PROTOCOL",
   +chainId: ProtocolSymbol,
 |};
 
 export type ChainId = EvmChainId | ProtocolSymbol;
-export type CurrencyId = EvmId | ProtocolId;
+export type Currency = Evm | Protocol;
 
 export const evmChainParser: C.Parser<EvmChainId> = C.fmap(
   C.string,
   parseEvmChainId
 );
 
-export const evmIdParser: C.Parser<EvmId> = C.object({
+export const evmParser: C.Parser<Evm> = C.object({
   type: C.exactly(["EVM"]),
   chainId: evmChainParser,
   currencyAddress: ethAddressParser,
```

Comment:
Much nicer :) Laaaaast nit would be to name this field `tokenAddress` or `tokenContract` instead of `currencyAddress` since that's the standard name used for this field in most web3 project APIs / docs etc.

e.g. from OpenSea JS: 
![image](https://user-images.githubusercontent.com/7143583/107682462-9d322680-6c5d-11eb-95b3-0095d408d551.png)

From Balancer:
![image](https://user-images.githubusercontent.com/7143583/107682620-c9e63e00-6c5d-11eb-8ccd-ae8792c9dba4.png)

From Uniswap:
![image](https://user-images.githubusercontent.com/7143583/107682781-fe59fa00-6c5d-11eb-8224-1fa8471f31fa.png)

Reply:
@hammadj I'd be happy to use `token` and `tokenAddress` but we have a special case for native ETH itself. (or xDAI on the xDai sidechain) and I want to be clear that we're accounting for those. We also have other special cases like SAI which don't conform to the ERC-20 standard and therefore begs the question: Is SAI actually a token if it's not ERC-20 conformant?